app/logic/CapiPrefiller.scala

```diff
@@ -0,0 +1,96 @@
+package logic
+
+import com.gu.contentapi.client.model.v1.{Content, TagType}
+import com.gu.facia.api.utils.{CardStyle, ResolvedMetaData}
+import com.gu.facia.client.models.TrailMetaData
+import model.editions.{Image, MediaType}
+import services.Prefill
+
+object CapiPrefiller {
+
+  private val pluralKickers = Set(
+    "tone/letters",
+    "tone/analysis",
+    "tone/obituaries"
+  )
+  private val singularKickers = Set(
+    "tone/reviews",
+    "tone/editorials",
+    "tone/matchreports",
+    "tone/explainers"
+  )
+  private val bylineKickers = Set(
+    "tone/comment"
+  )
+
+  def prefill(content: Content): Prefill = {
+    val internalPageCode = content.fields.flatMap(_.internalPageCode).getOrElse(-1)
+    val cardStyle = CardStyle(content, TrailMetaData.empty)
+    val metadata = ResolvedMetaData.fromContent(content, cardStyle)
+
+    // ResolvedMetaData here has a ton of booleans, but some may have been set naively
+    // If the cutout bool is set, we want to find the cutout values and add them in to a sub-object
+    val (_, maybeCutout) = CapiPrefiller.getFirstContributorWithCutoutOption(content, Some(metadata.imageCutoutReplace))
+
+    val (mediaType, cutoutImage) = (metadata.imageCutoutReplace, maybeCutout) match {
+      // if cutout desired and a cutout was found then configure cutout
+      case (true, Some(url)) => (Some(MediaType.Cutout), Some(Image(None, None, url, url)))
+      // if no cutout desired then default nothing
+      case (false, _) => (None, None)
+      // finally if a cutout is desired but nothing appropriate was found then explicitly set to use article trail
+      case (true, None) => (Some(MediaType.UseArticleTrail), None)
+    }
+    val newMetadata = metadata.copy(imageCutoutReplace = mediaType.contains(MediaType.Cutout))
+
+    val pickedKicker = pickKicker(content)
+
+    Prefill(
+      internalPageCode.toInt,
```

Comment:
Is it possible that the get on line 27 could return something that wouldn't parse as an Int?

Reply:
Good spot, but no, because after this refactoring, internal page code is ... an int.

I'll remove the toInt :)